presto-main/src/main/java/com/facebook/presto/operator/aggregation/BindableAggregationFunction.java

```diff
@@ -118,17 +104,45 @@ public InternalAggregationFunction specialize(BoundVariables variables, int arit
                 stateFactory,
                 outputType);
 
-        AccumulatorFactoryBinder factory = new LazyAccumulatorFactoryBinder(metadata, classLoader);
-
+        // Create specialized InternalAggregregationFunction for Presto
         return new InternalAggregationFunction(getSignature().getName(),
                 inputTypes,
-                intermediateType,
+                stateSerializer.getSerializedType(),
                 outputType,
                 details.isDecomposable(),
-                factory);
+                new LazyAccumulatorFactoryBinder(metadata, classLoader));
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return details.getDescription().orElse("");
+    }
+
+    private AggregationImplementation findMatchingImplementation(Signature boundSignature, BoundVariables variables, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        Optional<AggregationImplementation> implementation = Optional.empty();
+        if (implementations.getExactImplementations().containsKey(boundSignature)) {
+            implementation = Optional.of(implementations.getExactImplementations().get(boundSignature));
+        }
+        else {
+            for (AggregationImplementation genericImpl : implementations.getGenericImplementations()) {
```

Comment:
We generally avoid abbreviations, maybe call this `implementation` instead of `genericImpl`. (You can call the existing `implementation` `result`)